### ISSUE TITLE
Show std(out/err) when debugging

### ISSUE THREAD

**Issue by KotlinIsland** [robotframework/robotframework#3976]
When using PyCharm to debug robot keywords written in python, robot will capture all the console output.

This is really annoying, and am forced to use `logger.console()` when in the console, but even that isn't good enough because if there's an exception in the expression it won't print anything so I'm basically forced to use:

```py
try: logger.console(expr)
except Exception as e: logger.console(e)
```

![image](https://user-images.githubusercontent.com/65446343/118920618-1a1a1e00-b97a-11eb-9deb-13a282c56f37.png)
# 😲

**Comment by pekkaklarck** [comment]
Robot intercepting stdout/err and catching exceptions and showing all this information in the log file is a feature. That said, having a way to somehow tell the framework to just log everything to the console could be handy. Do you have ideas how it could work?

Notice that we already have somewhat related `--debugfile` and that it's possible to create listeners that log everything to console already now. Support for more verbose console output (#1393) is also related.

**Comment by KotlinIsland** [comment]
Listeners won't log things that are printed to stdout ie `print` or using the console.

Is there a way to capture the stdout and save it to the log but to also send it to the normal stdout?

**Comment by KotlinIsland** [comment]
Want me to put together a pr with this implementation?
```py
import sys
from typing import TextIO


class OutputDuplicator:
    def __enter__(self):
        self.out_file = TextIO()
        self.err_file = TextIO()
        self.stdout = sys.stdout
        self.stderr = sys.stderr
        sys.stdout = DuplicateWriter(self.out_file, sys.stdout)
        sys.stderr = DuplicateWriter(self.err_file, sys.stderr)

    def __exit__(self, a, b, c):
        sys.stdout = self.stdout
        sys.stderr = self.stderr
        self.out_file.close()
        self.err_file.close()


class DuplicateWriter:
    def __init__(self, file, stream):
        self.file = file
        self.stream = stream

    def write(self, data):
        self.file.write(data)
        self.stream.write(data)

    def flush(self):
        self.file.flush()


with OutputDuplicator() as o:
    print("asdf")
```

**Comment by KotlinIsland** [comment]
There's also the issue of console output being on the same line as the test name and pushing the result to the start of the next line. But this is a separate issue.
![image](https://user-images.githubusercontent.com/65446343/119914404-5a455600-bfa3-11eb-877e-74758bf1fd71.png)

**Comment by pekkaklarck** [comment]
I don't think the implementation above (and in PR #3988) is a good idea for these two important reasons:

- Printed messages should not be written to the console by default because most of the time they just add noise. Being able to enable that could be considered.
- The implementation only handles messages logged using `print()`, not messages logged using `robot.api.logger` or `logging`.

**Comment by pekkaklarck** [comment]
Notice that listeners do get all logged messages if they implement `log_message` method. A problem with messages logged using `print()` is that they are logged only after the keyword finishes, not when `print()` is actually called. That's due to our somewhat naïve stdout/stderr intercepting implementation what could possibly be enhanced.

**Comment by pekkaklarck** [comment]
No updates since June and I doubt this is a good idea in general. Closing.
